internal/rdc/client.go

```diff
@@ -0,0 +1,310 @@
+		URL:            url,
+		Username:       username,
+		AccessKey:      accessKey,
+		ArtifactConfig: artifactConfig,
+	}
+}
+
+// ReadAllowedCCY returns the allowed (max) concurrency for the current account.
+func (c *Client) ReadAllowedCCY(ctx context.Context) (int, error) {
+	req, err := requesth.NewWithContext(ctx, http.MethodGet,
+		fmt.Sprintf("%s/v1/rdc/concurrency", c.URL), nil)
+	if err != nil {
+		return 0, err
+	}
+	req.SetBasicAuth(c.Username, c.AccessKey)
+
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return 0, fmt.Errorf("unexpected statusCode: %v", resp.StatusCode)
+	}
+
+	var cr concurrencyResponse
+	if err := json.NewDecoder(resp.Body).Decode(&cr); err != nil {
+		return 0, err
+	}
+	return cr.Organization.Maximum, nil
+}
+
+// ReadJob returns the job details.
+func (c *Client) ReadJob(ctx context.Context, id string) (job.Job, error) {
+	req, err := requesth.NewWithContext(ctx, http.MethodGet,
+		fmt.Sprintf("%s/v1/rdc/jobs/%s", c.URL, id), nil)
+	if err != nil {
+		return job.Job{}, err
+	}
+	req.SetBasicAuth(c.Username, c.AccessKey)
+
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return job.Job{}, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return job.Job{}, fmt.Errorf("unexpected statusCode: %v", resp.StatusCode)
+	}
+
+	var jr readJobResponse
+	if err := json.NewDecoder(resp.Body).Decode(&jr); err != nil {
+		return job.Job{}, err
+	}
+	return job.Job{
+		ID:     id,
+		Error:  jr.Error,
+		Status: jr.Status,
+		Passed: jr.Status == job.StatePassed,
+	}, nil
+}
+
+// PollJob polls job details at an interval, until the job has ended, whether successfully or due to an error.
+func (c *Client) PollJob(ctx context.Context, id string, interval time.Duration) (job.Job, error) {
+	req, err := requesth.NewWithContext(ctx, http.MethodGet,
+		fmt.Sprintf("%s/v1/rdc/jobs/%s", c.URL, id), nil)
+	if err != nil {
+		return job.Job{}, err
+	}
+	req.SetBasicAuth(c.Username, c.AccessKey)
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		j, err := doRequestStatus(c.HTTPClient, req)
+		if err != nil {
+			return job.Job{}, err
+		}
+
+		if job.Done(j.Status) {
+			j.IsRDC = true
+			return j, nil
+		}
+	}
+
+	return job.Job{}, nil
+}
+
+func doRequestStatus(httpClient *http.Client, request *http.Request) (job.Job, error) {
+	resp, err := httpClient.Do(request)
+	if err != nil {
+		return job.Job{}, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= http.StatusInternalServerError {
+		return job.Job{}, ErrServerError
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return job.Job{}, ErrJobNotFound
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		err := fmt.Errorf("job status request failed; unexpected response code:'%d', msg:'%v'", resp.StatusCode, string(body))
+		return job.Job{}, err
+	}
+
+	jobDetails := job.Job{}
```

Comment:
More canonical if you are just declaring the variable:
```suggestion
	var jobDetails job.Job
```